File: python/tests/test_describe_namespace.py
import pytest

from bedrock_bio.describe_namespace import describe_namespace
from bedrock_bio.list_tables import list_tables


class TestDescribeNamespace:
    def test_errors_on_unknown_namespace(self, v2_manifest):
        with pytest.raises(ValueError, match="not found"):
            describe_namespace("not_a_namespace")

    def test_returns_expected_fields(self, v2_manifest):
        result = describe_namespace("test_ns")
        assert result["name"] == "Test Namespace"
        assert isinstance(result["license"], str)
        assert isinstance(result["citation"], str)
        assert isinstance(result["context"], str)

    def test_tables_field_is_list_of_fqns(self, v2_manifest):
        result = describe_namespace("test_ns")
        assert len(result["tables"]) > 0
        assert all(fqn.startswith("test_ns.") for fqn in result["tables"])
        assert "test_ns.test_tbl" in result["tables"]

    def test_tables_agree_with_list_tables_filtered(self, v2_manifest):
        result = describe_namespace("test_ns")
        from_list = [t for t in list_tables() if t.startswith("test_ns.")]
        assert set(result["tables"]) == set(from_list)

File: r/tests/testthat/test-describe_namespace.R
test_that("errors on unknown namespace", {
  local_v2_manifest()
  expect_error(
    describe_namespace("not_a_namespace"),
    "not found"
  )
})

test_that("returns expected fields", {
  local_v2_manifest()
  result <- describe_namespace("test_ns")
  expect_equal(result$name, "Test Namespace")
  expect_type(result$name, "character")
  expect_true(nzchar(result$name))
  expect_type(result$license, "character")
  # citation is a pre-formatted string in v2.
  expect_type(result$citation, "character")
  expect_type(result$context, "character")
})

test_that("tables field is character vector of fully-qualified names", {
  local_v2_manifest()
  result <- describe_namespace("test_ns")
  expect_type(result$tables, "character")
  expect_true(length(result$tables) > 0)
  expect_true(all(startsWith(result$tables, "test_ns.")))
  expect_true("test_ns.test_tbl" %in% result$tables)
})

test_that("tables list agrees with list_tables filtered to namespace", {
  local_v2_manifest()
  result <- describe_namespace("test_ns")
  from_list <- list_tables()[startsWith(list_tables(), "test_ns.")]
  expect_setequal(result$tables, from_list)
})
